src/griptape_nodes/exe_types/param_components/artifact_url/public_artifact_url_parameter.py

```diff
@@ -0,0 +1,155 @@
+    supported_artifact_type_names: ClassVar[list[str]] = [cls.__name__ for cls in supported_artifact_types]
+    gtc_file_path: Path | None = None
+
+    def __init__(
+        self, node: BaseNode, artifact_url_parameter: Parameter, disclaimer_message: str | None = None
+    ) -> None:
+        self._node = node
+        self._parameter = artifact_url_parameter
+        self._disclaimer_message = disclaimer_message
+
+        if artifact_url_parameter.type.lower() not in [name.lower() for name in self.supported_artifact_type_names]:
+            msg = (
+                f"Unsupported artifact type '{artifact_url_parameter.type}' for "
+                f"artifact URL parameter '{artifact_url_parameter.name}'. "
+                f"Supported types: {', '.join(self.supported_artifact_type_names)}"
+            )
+            raise ValueError(msg)
+
+        api_key = str(self._get_secret_value(self.API_KEY_NAME))
+        base = os.getenv("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")
+        self._storage_driver = GriptapeCloudStorageDriver(
+            workspace_directory=GriptapeNodes.ConfigManager().workspace_path,
+            bucket_id=self._get_bucket_id(base, api_key),
+            api_key=api_key,
+            base_url=base,
+        )
+
+    @classmethod
+    def _get_bucket_id(cls, base_url: str, api_key: str) -> str:
+        bucket_id: str | None = cls._get_secret_value(cls.BUCKET_ID_NAME, should_error_on_not_found=False)
+
+        if bucket_id is not None:
+            return bucket_id
+
+        buckets = GriptapeCloudStorageDriver.list_buckets(
+            base_url=base_url,
+            api_key=api_key,
+        )
+        if len(buckets) == 0:
+            msg = "No Griptape Cloud storage buckets found!"
+            raise RuntimeError(msg)
+
+        return buckets[0]["bucket_id"]
+
+    @classmethod
+    def _get_config_value(cls, key: str, default: Any | None = None) -> Any | None:
+        request = GetConfigValueRequest(category_and_key=key)
+        result_event = GriptapeNodes.handle_request(request)
+
+        if isinstance(result_event, GetConfigValueResultSuccess):
+            return result_event.value
+
+        return default
+
+    @classmethod
+    def _get_secret_value(
+        cls, key: str, default: Any | None = None, *, should_error_on_not_found: bool = False
+    ) -> Any | None:
+        request = GetSecretValueRequest(key=key, should_error_on_not_found=should_error_on_not_found)
+        result_event = GriptapeNodes.handle_request(request)
+
+        if isinstance(result_event, GetSecretValueResultSuccess):
+            return result_event.value
+
+        return default
+
+    def add_input_parameters(self) -> None:
+        self._node.add_node_element(
+            ParameterMessage(
+                name=f"artifact_url_parameter_message_{self._parameter.name}",
+                title="Artifact URL Upload",
```

Comment:
Do we need the word "Artifact" in here? customers don't really know what an Artifact is.. could it be Media URL Upload?